main.tf

```
@@ -608,7 +608,7 @@ resource "aws_ecs_service" "atlantis" {
     }
   }
 
-  tags = local.tags
+  tags = var.use_old_arn_format ? null : local.tags
```

I don't have any way of testing since I don't have any accounts that do not use the newer format, but what if we do this instead (related to this [comment](https://github.com/terraform-aws-modules/terraform-aws-atlantis/pull/153#issuecomment-703212662))
```suggestion
  tags = var.use_old_arn_format ? {} : local.tags
```

---

It's always better to do `null` since that safely removes the argument from being set in the first place.

We've been using this method in the https://github.com/cloudposse/terraform-aws-ecs-alb-service-task module with positive results.

---

I don't believe it always removes, which is why I was suggesting an empty map which removes any ambiguity. I do agree that `null` works great as a default variable value, but this is something slightly different - a different edge case. either way, we can try this approach. one thing though - I think we should adjust the variable name to `use_ecs_old_arn_format` to make it a bit more clear

---

> I don't believe it always removes

I tested this with 0.12.x, 0.13.x, and 0.14.x and it seems to remove the argument completely from the resource.

> either way, we can try this approach.

Ok 😄 

> one thing though - I think we should adjust the variable name to use_ecs_old_arn_format to make it a bit more clear

Agreed, merge conflicts fixed and that change was committed.